src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -351,23 +343,72 @@ private bool HasJobAttribute(CustomAttributeData attributeData)
             };
         }
 
-        private class ListenerFactory : IListenerFactory
+        internal class ListenerFactory : IListenerFactory
         {
             private readonly FunctionDescriptor _descriptor;
             private readonly ITriggeredFunctionExecutor _executor;
             private readonly ITriggerBinding _binding;
+            private readonly TraceWriter _trace;
 
-            public ListenerFactory(FunctionDescriptor descriptor, ITriggeredFunctionExecutor executor, ITriggerBinding binding)
+            public ListenerFactory(FunctionDescriptor descriptor, ITriggeredFunctionExecutor executor, ITriggerBinding binding, TraceWriter trace)
             {
                 _descriptor = descriptor;
                 _executor = executor;
                 _binding = binding;
+                _trace = trace;
             }
 
-            public Task<IListener> CreateAsync(CancellationToken cancellationToken)
+            public async Task<IListener> CreateAsync(CancellationToken cancellationToken)
             {
                 ListenerFactoryContext context = new ListenerFactoryContext(_descriptor, _executor, cancellationToken);
-                return _binding.CreateListenerAsync(context);
+                IListener listener = await _binding.CreateListenerAsync(context);
+                return new TriggerListener(listener, _descriptor, _trace);
+            }
+
+            private class TriggerListener : IListener
```

Comment:
>TriggerListener [](start = 26, length = 15)

1. There's a lot of listeners scattered throughout the code.  Is this the only one that needs to be wrapped?
2. Would it be simpler to try/catch the place that calls listener.start() instead of creating a wrapper?